iree/base/file_io.h

```diff
@@ -45,6 +46,14 @@ Status DeleteFile(const std::string& path);
 Status MoveFile(const std::string& source_path,
                 const std::string& destination_path);
 
+// Gets a platform and environment-dependent path for temporary files.
+std::string GetTempPath();
+
+// Gets a temporary file name and returns its absolute path.
+// The particular path chosen is platform and environment-dependent.
+// Unique characters will be automatically inserted after |base_name|.
+StatusOr<std::string> GetTempFile(const std::string& base_name);
```

Comment:
string view here too?

Reply:
Yes, this one can be `string_view`. I looked at the others and a few of the underlying system calls need c-style strings that a null-terminated, so `string_view` isn't appropriate for them.

I'll make this change as part of merging, and not on the GitHub PR here (found a really weird bug while pulling this in, updating the PR will just be busy work).